File: apps/ios/ScribbleFit/Sources/Domain/UseCases/GetScribblesForDateUseCase.swift
import Foundation

@MainActor
public struct GetScribblesForDateUseCase: Sendable {
    private let repository: ScribbleRepository

    public init(repository: ScribbleRepository) {
        self.repository = repository
    }

    public func execute(date: Date) -> AsyncStream<[Scribble]> {
        repository.getScribbles(for: date)
    }
}

File: apps/android/feature/scribble/domain/src/main/java/com/scribblefit/feature/scribble/domain/usecase/GetScribblesForDateUseCase.kt
package com.scribblefit.feature.scribble.domain.usecase

import com.scribblefit.core.model.CurrentDate
import com.scribblefit.core.model.Scribble
import com.scribblefit.feature.scribble.domain.ScribbleRepository
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOn

class GetScribblesForDateUseCase(
    private val repository: ScribbleRepository,
    private val coroutineDispatcher: CoroutineDispatcher
) {
    operator fun invoke(date: CurrentDate): Flow<List<Scribble>> {
        return repository.getScribblesByDate(date.millis).flowOn(coroutineDispatcher)
    }
}